dts/bindings/pinctrl/ambiq,apollo3-pinctrl.yaml

```diff
@@ -0,0 +1,122 @@
+  The Ambiq Apollo3 pin controller is a node responsible for controlling
+  pin function selection and pin properties, such as routing a UART0 TX
+  to pin 60 and enabling the pullup resistor on that pin.
+
+  The node has the 'pinctrl' node label set in your SoC's devicetree,
+  so you can modify it like this:
+
+  &pinctrl {
+    /* your modifications go here */
+  };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  /* You can put this in places like a board-pinctrl.dtsi file in
+   * your board directory, or a devicetree overlay in your application.
+  */
+
+  /* include pre-defined combinations for the SoC variant used by the board */
+  #include <dt-bindings/pinctrl/ambiq-apollo3-pinctrl.h>
+
+  &pinctrl {
+    uart0_default: uart0_default {
+      group1 {
+        pinmux = <UART0TX_P60>;
+      };
+      group2 {
+        pinmux = <UART0RX_P47>;
+        input-enable;
+      };
+    };
+  };
+
+  The 'uart0_default' child node encodes the pin configurations for a
+  particular state of a device; in this case, the default (that is, active)
+  state.
+
+  As shown, pin configurations are organized in groups within each child node.
+  Each group can specify a list of pin function selections in the 'pinmux'
+  property.
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'input-enable' property in group 2.
+
+compatible: "ambiq,apollo3-pinctrl"
+
+include: base.yaml
+
+child-binding:
+  description: |
+    Definitions for a pinctrl state.
+  child-binding:
+
+    include:
+      - name: pincfg-node.yaml
+        property-allowlist:
+          - input-enable
+          - drive-push-pull
+          - drive-open-drain
+          - bias-high-impedance
+          - bias-pull-up
+          - bias-pull-down
+
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of pins sharing the same group properties. Each
+          element of the array is an integer constructed from the
+          pin number and the alternative function of the pin.
+      drive-strength:
+        type: string
+        enum:
+          - "0.1"
+          - "0.5"
+          - "0.75"
+          - "1.0"
+        default: "0.1"
+        description: |
+          The drive strength of a pin, relative to full-driver strength.
+          The default value is 0.1, which is the lowest one: 2mA.
+      ambiq,pull-up-ohms:
+        type: int
+        enum:
+          - 1500
+          - 6000
+          - 12000
+          - 24000
+        default: 1500
+        description: |
+          The 1.5K-24K pullup values are valid for select I2C enabled pads.
+          For Apollo3 these pins are 0-1,5-6,8-9,25,27,39-40,42-43,48-49.
+          The default value is 1500 ohms, which is the reset value of
+          register PADREGx.PADxRSEL.
+      ambiq,iom-nce-module:
+        type: int
+        default: 0
```

Comment:
There are many nxp, st, atmel that currently have default:.
It is not in the Zephyr documentation anywhere that default: cannot or should not be used.

https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr%20default%3A&type=code

Reply:
every single case conceivable cannot be documented

In this case I am not sure it makes sense to have a default value for a chip select, why do you think it does?

Reply:
Yes, and I am asking why is your judgement that it should have a default value instead of required: true, since specifying what is the chip select seems like it would make sense to make the user make an explicit choice, but I am asking you this question because you authored the PR, I don't know

Reply:
The lines in italics are my response as to why; hence, it is valid.

Reply:
I am confused what you are meaning, are you saying it is a reset value? if so, just put that in the description? or what is the unusual condition?

Reply:
I answered your question exactly as asked.

> Driver developers should use their best judgment

We looked at the documentation and chose the best option for the Ambiq code.
We chose to use the default value and according to the documentation, that is our choice to make.

Reply:
You can put the default value, if you put a justification in the description, which I still don't see. Otherwise put the dev review label on this PR because I am tired of this pointless back and forth

Reply:
Justification added